crates/web_server/src/lib.rs

```diff
@@ -24,24 +28,19 @@ type Error = ServerError;
 /// Data is stored as a `Vec<u8>` to not incur overhead from deserializing
 struct WebServerState<KEY> {
     /// view number -> (secret, proposal)
-    proposals: HashMap<u64, (String, Vec<u8>)>,
-
-    view_sync_proposals: HashMap<u64, Vec<(u64, Vec<u8>)>>,
-
-    view_sync_proposal_index: HashMap<u64, u64>,
+    proposals: BTreeMap<u64, (String, Vec<u8>)>,
+    /// for view sync: view number -> (relay, certificate)
+    view_sync_certificates: BTreeMap<u64, Vec<(u64, Vec<u8>)>>,
+    /// view number -> relay
+    view_sync_certificate_index: HashMap<u64, u64>,
     /// view number -> (secret, da_certificates)
     da_certificates: HashMap<u64, (String, Vec<u8>)>,
-    /// view for oldest proposals in memory
-    oldest_proposal: u64,
     /// view for the most recent proposal to help nodes catchup
     latest_quorum_proposal: u64,
     /// view for the most recent view sync proposal
-    latest_view_sync_proposal: u64,
-
-    /// view for teh oldest DA certificate
+    latest_view_sync_certificate: u64,
+    /// view for the oldest DA certificate
```

Comment:
Whoops missed this in the last one, we can get rid of this too, but let's keep it for now.